mmtk/src/scanning.rs

```diff
@@ -151,8 +144,36 @@ impl Scanning<JikesRVM> for VMScanning {
     }
 }
 
+struct ObjectsClosure<'a, E: ProcessEdgesWork<VM = JikesRVM>>(Vec<Address>, &'a mut GCWorker<JikesRVM>, PhantomData<E>);
```

Comment:
We also have an `ObjectsClosure` in OpenJDK. I wonder if we could refactor it to mmtk-core at some point. It seems VM neutral.